src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -1697,6 +1697,56 @@ static QVariant fcnRasterValue( const QVariantList &values, const QgsExpressionC
   return std::isnan( value ) ? QVariant() : value;
 }
 
+static QVariant fcnRasterAttributes( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsRasterLayer *layer = QgsExpressionUtils::getRasterLayer( values.at( 0 ), parent );
+  if ( !layer || !layer->dataProvider() )
+  {
+    parent->setEvalErrorString( QObject::tr( "Function `raster_attributes` requires a valid raster layer." ) );
+    return QVariant();
+  }
+
+  int bandNb = QgsExpressionUtils::getNativeIntValue( values.at( 1 ), parent );
+  if ( bandNb < 1 || bandNb > layer->bandCount() )
+  {
+    parent->setEvalErrorString( QObject::tr( "Function `raster_attributes` requires a valid raster band number." ) );
+    return QVariant();
+  }
+
+  const double value = QgsExpressionUtils::getDoubleValue( values.at( 2 ), parent );
+  if ( std::isnan( value ) )
+  {
+    parent->setEvalErrorString( QObject::tr( "Function `raster_attributes` requires a valid raster value." ) );
+    return QVariant();
+  }
+
+  if ( ! layer->dataProvider()->attributeTable( bandNb ) )
+  {
+    return QVariant();
+  }
+
+  const QVariantList data = layer->dataProvider()->attributeTable( bandNb )->row( value );
+  if ( data.isEmpty() )
+  {
+    return QVariant();
+  }
+
+  QVariantMap result;
+  const QList<QgsRasterAttributeTable::Field> fields { layer->dataProvider()->attributeTable( bandNb )->fields() };
+  for ( int idx = 0; idx < static_cast<int>( fields.count( ) ) && idx < static_cast<int>( data.count() ); ++idx )
```

Comment:
Why the static_cast ?

Reply:
Because of QT6: https://doc.qt.io/qt-6/qlist.html#count-1 it returns a qsizetype (size_t), I had big troubles on windows due to this missing cast (crashes).

Reply:
and if we change `idx` type to `size_t`, does it solve the issue ?